mne/preprocessing/tests/test_stim.py

```diff
@@ -25,52 +27,50 @@ def test_stim_elim():
     tidx = int(events[event_idx, 0] - raw.first_samp)
 
     # use window around stimulus
-    tmin = -0.02
-    tmax = 0.02
-    test_tminidx = int(-0.01 * raw.info['sfreq'])
-    test_tmaxidx = int(0.01 * raw.info['sfreq'])
+    tmin, tmax = -0.02, 0.02
+    tmin_samp = int(-0.01 * raw.info['sfreq'])
+    tmax_samp = int(0.01 * raw.info['sfreq'])
 
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
-                                  tmax=tmax, mode='linear')
-    data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
+    raw = fix_stim_artifact(raw, events, event_id=1, tmin=tmin,
+                                tmax=tmax, mode='linear')
+    data, times = raw[:, (tidx + tmin_samp):(tidx + tmax_samp)]
     diff_data0 = np.diff(data[0])
     diff_data0 -= np.mean(diff_data0)
     assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
-                                  tmax=tmax, mode='window')
-    data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
+    raw = fix_stim_artifact(raw, events, event_id=1, tmin=tmin,
+                                tmax=tmax, mode='window')
+    data, times = raw[:, (tidx + tmin_samp):(tidx + tmax_samp)]
     assert_true(np.all(data) == 0.)
 
     # use window before stimulus
-    tmin = -0.045
-    tmax = 0.015
-    test_tminidx = int(-0.035 * raw.info['sfreq'])
-    test_tmaxidx = int(-0.015 * raw.info['sfreq'])
+    tmin, tmax, event_id = -0.2, 0.5, 1
+    picks = pick_types(raw.info, meg=True, eeg=True,
+                       eog=True, stim=False, exclude='bads')
+    epochs = Epochs(raw, events, event_id, tmin, tmax,
+                    picks=picks, preload=True)
+    e_start = int(np.ceil(epochs.info['sfreq'] * epochs.tmin))
+    tmin, tmax = -0.045, -0.015
+    tmin_samp = int(-0.035 * epochs.info['sfreq']) - e_start
+    tmax_samp = int(-0.015 * epochs.info['sfreq']) - e_start
 
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
-                                  tmax=tmax, mode='linear')
-    data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
-    diff_data0 = np.diff(data[0])
+    epochs = fix_stim_artifact(epochs, None, None, tmin, tmax, mode='linear')
+    data = epochs.get_data()[:, :, tmin_samp:tmax_samp]
+    diff_data0 = np.diff(data[0][0])
     diff_data0 -= np.mean(diff_data0)
     assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
-                                  tmax=tmax, mode='window')
-    data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
+    epochs = fix_stim_artifact(epochs,None, None, tmin, tmax, mode='window')
```

Comment:
if `None` is some default parameter, you don't need to pass it. Use keyword arguments to pass only what is required